orchestrator/utils/search_query.py

```diff
@@ -287,16 +312,29 @@ def visit_kv_term(node: Node, acc: list[str]) -> None:
 
     @staticmethod
     def visit_search_word(node: Node, acc: list[str]) -> None:
+
+        if node[0] not in ["Word", "PrefixWord"]:
+            raise Exception(f"Invalid SearchWord type {node[0]}")
+
         # Postgres is finicky with single quotes in the query. In some situations it throws an error.
         # To avoid the special handling of ' by PG, we replace it with double quotes.
-        # We also replace underscores with the 'followed by' operator
-        text = node[1].replace("'", '"').replace("_", " <-> ")
-        if node[0] == "Word":
-            acc.append(text)
-        elif node[0] == "PrefixWord":
-            acc.append(f"{text}:*")
-        else:
-            raise Exception(f"Invalid SearchWord type {node[0]}")
+
+        text = node[1].replace("'", '"')
+
+        parts = TSQueryVisitor._split_term(node[1])
+
+        if not parts:
+            return
+
+        if len(parts) > 1:
+            text = " <-> ".join(parts)
```

Comment:
The len check is not needed probably?

```suggestion
        text = " <-> ".join(parts)
```